pandas/core/internals/managers.py

```diff
@@ -334,7 +334,7 @@ def reduce(self: T, func) -> T:
         # If 2D, we assume that we're operating column-wise
         assert self.ndim == 2
 
-        res_blocks = []
+        res_blocks: List[Block] = []
```

Comment:
mypy doesn't need this if refactored as 

```
res_blocks = [i for blk in self.blocks for i in blk.reduce(func)]
```

Revealed type is 'builtins.list[pandas.core.internals.blocks.Block*]

again, it's down to a personal preference of variable type annotations vs (in some cases complex) list comprehensions

Reply:
I've got a couple other PRs that refactor this method, so not going to worry about this too much

Reply:
sure